### DISCUSSION TITLE
Yolov8 Segmentation: Mask as Label

### DISCUSSION THREAD

**Discussion by stevenzielke** [ultralytics/ultralytics#871] [General]
Hi Community.
Is it possible to use a mask as a label for segmentation? Polygon is not enough because the mask has enclosed features and is pretty complex. Areas need also to be excluded from the segmentation. We have been using RLE on other projects but I cannot find a way to convert RLE labels to YOLO labels.

Thanks

**Comment by glenn-jocher** [comment]
@stevenzielke hello!

Yes, YOLOv8 supports segmentation tasks, and you can indeed use masks as labels for segmentation. While YOLO traditionally uses bounding boxes for object detection, YOLOv8 extends its capabilities to handle more complex annotations like masks for segmentation tasks.

For your specific case with RLE (Run-Length Encoding) labels, you would need to convert them into a format compatible with YOLOv8. This typically involves creating binary masks from your RLE labels and then following the segmentation format outlined in our documentation.

If you have masks in RLE format, you'll need to preprocess them into the required format for training with YOLOv8. This preprocessing step is not directly handled by YOLOv8, but once you have your masks ready, you can proceed with training your segmentation model.

For further guidance on how to format your segmentation labels and use them for training, please refer to the segmentation section in our documentation. 😊📚

Best of luck with your segmentation project!
